app/pages/_app.js

```diff
@@ -5,15 +5,23 @@ import SSRProvider from 'react-bootstrap/SSRProvider';
 import {SessionProvider} from 'next-auth/react';
 import { ApolloProvider } from '@apollo/client';
 import client from '../apollo-client';
+import { useState } from 'react';
+import { Router } from 'next/router';
 
 function MyApp({ Component, pageProps: {session, ...pageProps}}) {
+  const [loading, setLoading] = useState(false);
+
+  Router.events.on('routeChangeStart', () => setLoading(true))
+  Router.events.on('routeChangeComplete', () => setLoading(false))
+
```

Comment:
Could we not do this inside the `<Layout />`?

Reply:
Thanks , I have applied the changes